kerastuner/tuners/ultraband/new_ultraband.py

```diff
@@ -0,0 +1,65 @@
+import queue
+
+from kerastuner.engine import Tuner
+from kerastuner.engine.oracle import Oracle
+
+
+class UltraBandTuner(Tuner):
+    def run_trial(self, hp, trial_id, *fit_args, **fit_kwargs):
+        model = self._build_model(hp)
+        trial = self._add_trial(model, hp)
+        fit_kwargs['epochs'] = hp.values['tuner/epochs']
+        trial.run_execution(*fit_args, **fit_kwargs)
+        result = trial.best_metrics[self.objective]
+        self.oracle.result(hp.values, result)
+
+
+class UltraBandOracle(Oracle):
+
+    def __init__(self, trials):
+        super().__init__()
+        self.trials = trials
+        self.queue = queue.Queue()
+        self.during_batch = False
+        self._first_time = True
+        self._perf_record = {}
+        self._current_round = 0
+        self.spaces = []
+        self._model_sequence = []
+
+    def result(self, trial_id, score):
+        self._perf_record[trial_id] = score
+
+    def update_space(self, additional_hps):
+        pass
+
+    def populate_space(self, trial_id, space):
+        # dispatch new trials with low epochs
+        # occasionally re-run pretty good trials with more epochs
+        # if run out of trials, IDLE
+        # when only one model left, EXIT
```

Comment:
To confirm: is the Oracle tracking trials that have been handed out but not completed? I think the logic is really:

```
if we're out of trials:
    if we know there are still trials processing:
        IDLE  # We're waiting for the current round to finish, but will have more work
                   # to dispatch after it does, or sooner if workers die.
   else:
        EXIT  # We're done.
```